### PR TITLE
[RAINCATCH-1258] UI guide

### PR DESCRIPTION
**Pull request by paolobueno**
- [X] Mention technologies: Angular, Bootstrap
- [X] Describe how projects are formatter.
- [x] How people can add new elements to UI. 
- [x] How people can modify styles.
- [x] Adding their own categories to navigator.

### CODE REVIEW

File: docs/workforce-management-framework/topics/con-angularjs-modules.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
+= {WFM-RC-NameShort} AngularJS modules
```
Comment (austincunningham) [inline docs/workforce-management-framework/topics/con-angularjs-modules.adoc]: need to rebase master {WFM-RC-NameShort} is now {Feature}

File: docs/workforce-management-framework/topics/pro-ui-customization.adoc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+. Utilize the UI-Router's `$stateProvider` to include any new routes
+. These new routes should be child routes to `app.`, since otherwise the views in the custom route will replace the application's entire interface instead of inserting their content in the containers
+
+Refer to the link:https://docs.angularjs.org/api/ng/function/angular.module[AngularJS Module documentation] for further information on creating new modules.
+
+=== Adding the new module to the main application config
+
+[source,javascript]
+---
+---
```
Comment (austincunningham) [inline docs/workforce-management-framework/topics/pro-ui-customization.adoc]: WIP

File: docs/workforce-management-framework/topics/ref-demo-app.adoc [hunk 3]
```diff
@@ -25,3 +25,30 @@ Login using the following credentials:
 |`max`
 |`123`
 |===
+
+[id=-'{context}-structure-for-the-mobile-and-portal-starter-applications']
+ifdef::upstream[]
+discrete
+endif::upstream[]
```
Comment (austincunningham) [inline docs/workforce-management-framework/topics/ref-demo-app.adoc]: getting rid of the discrete now we have a dynamic toc

File: docs/workforce-management-framework/topics/ref-demo-app.adoc [hunk 4]
```diff
@@ -25,3 +25,30 @@ Login using the following credentials:
...
+== Structure for the Mobile and Portal Starter Applications
+
+The UI Starter Applications present a consistent directory structure that can serve as a sensible base for customizations.
+
+The following table highlights the function of the main folders:
+
+|==
```
Comment (austincunningham) [inline docs/workforce-management-framework/topics/ref-demo-app.adoc]: Issue with the table render 

image

**Comment by wtrocki** [comment]
@paolobueno Great work so far. Everything makes sense. I will wait for the rest to do final review

File: docs/workforce-management-framework/topics/ref-demo-app.adoc (hunk 4 above)
Comment (austincunningham) [reply docs/workforce-management-framework/topics/ref-demo-app.adoc]: needs an extra =
```|===```

File: docs/workforce-management-framework/topics/con-ui-technology.adoc [hunk 5]
```diff
@@ -0,0 +1,18 @@
+[id='{context}-ref-ui-technology']
+= The Technology and Structure of the User Interface in a {PROJECT_NAME} Project
+
+The User Interface of {PROJECT_NAME} consists of a series of modules made available through the public NPM registry and built as modules the link:https://angularjs.org/[AngularJS Framework]. 
+
+=== UI Frameworks
+
+The {PROJECT_NAME} modules are intended to be used as part of a regular AngularJS application and used by it as link:https://docs.angularjs.org/guide/module[AngularJS modules] they are kept in a separate, framework-specific repository on link:https://github.com/feedhenry-raincatcher/raincatcher-angularjs/[GitHub]
```
Comment (wtrocki) [inline docs/workforce-management-framework/topics/con-ui-technology.adoc]: Users may not need to know where they are kept- We have separate contributor guide that we going to use for this.

Comment (wtrocki) [reply docs/workforce-management-framework/topics/con-ui-technology.adoc]: Just lets avoid confusion from end users checking this repo.

Comment (paolobueno) [reply docs/workforce-management-framework/topics/con-ui-technology.adoc]: Great, I wasn't sure about this part, any links required?

Going to reword to just mention the framework itself.

Comment (wtrocki) [reply docs/workforce-management-framework/topics/con-ui-technology.adoc]: No links required as people will get applications depending on their support level upstream etc.

**Review by wtrocki — APPROVED** [review]

**Comment by wtrocki** [comment]
@jstaffor Ready for action!

**Comment by paolobueno** [comment]
@wtrocki changed target to `upstream`, could be okay to merge now :pray:

**Pull request merged**